packages/lodestar/src/db/seenSyncCommitteeContributionCache.ts

```diff
@@ -0,0 +1,90 @@
+import {phase0, altair} from "@chainsafe/lodestar-types";
+import {NUM_SLOTS_IN_CACHE, slotRootKey, SlotRootKey} from "./repositories/utils/syncCommittee";
+
+type ContriButionsBySlotRoot = Map<SlotRootKey, altair.SyncCommitteeContribution[]>;
+type SeenCacheKey = string;
+
+/**
+ * Cache SyncCommitteeContribution and seen ContributionAndProof.
+ * This is used for SignedContributionAndProof validation and block factory.
+ * This stays in-memory and should be pruned per slot.
+ */
+export class SeenSyncCommitteeContributionCache {
```

Comment:
Would it make sense to merge the `SeenSyncCommitteeContributionCache` and `SeenSyncCommitteeCache` into a single class? You would only need to namespace the `seenCache` maps

Reply:
I'd like to keep it separated because it cache is for different purposes respective to 2 separate gossip validations.
+ SeenSyncCommitteeCache has `contributionCaches` as an array by subnet and each one is a map, it's for `sync_committee_{subnet_id}` topic
+ SeenSyncCommitteeContributionCache has `contributionsCache` has a map by slot and `Map<SlotRootKey, altair.SyncCommitteeContribution[]>`, it's for sync_committee_contribution_and_proof

merging those to a single cache would make it confused to me when I want to look back